studio/src/mas-fragment-editor.js

```diff
@@ -1234,9 +1238,15 @@ export default class MasFragmentEditor extends LitElement {
         const previewFragment = this.fragmentStore?.previewStore?.value;
         prepopulateFragmentCache(this.fragment.id, previewFragment);
 
-        const masService = getService();
-        masService.setAttribute('locale', Store.localeOrRegion());
-        masService.activate();
+        document.querySelectorAll('mas-commerce-service').forEach((el) => el.remove());
+        const masServiceElement = document.createElement('mas-commerce-service');
```

Comment:
instead of doing this, I recommend to cause a re-render of the shadow dom.
ideally, the locale should be a store value.
when updated, a subscription updates the mas-commerce-service
and observer mas-fragment-editor.js would re-render after it.
assuming subscription is registered earlier.